src/interceptors/WebSocket/WebSocketServerConnection.ts

```diff
@@ -1,12 +1,23 @@
 import { invariant } from 'outvariant'
-import { kClose, WebSocketOverride } from './WebSocketOverride'
+import {
+  kClose,
+  WebSocketEventListener,
+  WebSocketOverride,
+} from './WebSocketOverride'
 import type { WebSocketData } from './WebSocketTransport'
 import type { WebSocketClassTransport } from './WebSocketClassTransport'
 import { bindEvent } from './utils/bindEvent'
 import { CancelableMessageEvent, CloseEvent } from './utils/events'
 
 const kEmitter = Symbol('kEmitter')
 
+interface WebSocketServerEventMap {
```

Comment:
Explicit event map so there can be a proper overlap of events in a `WebSocketClientConnection | WebSocketServerConnection` union.